torch/_dynamo/eval_frame.py

```diff
@@ -166,6 +166,8 @@ def _callback_from_stance(callback):
     elif _stance.stance == "eager_on_recompile":
         # run mode
         return False
+    elif _stance.stance == "skip_guard_eval":
+        return torch._C._dynamo.eval_frame.skip_guard_eval_callback_flag
```

Comment:
I'm ok with this idea in general, but this seems like a bit of an odd way to do this, since it will also disable compiling of new frames.  Maybe this should instead be a global flag in C++ that you can control without changing stances.

We should also put unsafe in the name, and I like the idea of something softer that still checks shapes.